src/app/nodetool/qgsmaptoolnodetool.cpp

```diff
@@ -630,7 +630,9 @@ void QgsMapToolNodeTool::canvasDoubleClickEvent( QgsMapMouseEvent* e )
   vlayer->beginEditCommand( tr( "Inserted vertex" ) );
 
   // add vertex
-  vlayer->insertVertex( layerCoords.x(), layerCoords.y(), mSelectedFeature->featureId(), snapResults.first().afterVertexNr );
+  QgsPointV2 p( layerCoords.x(), layerCoords.y() );
+  p.addZValue( defaultZValue() );
```

Comment:
IMO the 3rd value in QgsPointV2 ctor should be Z coordinate. Or implicit conversion from {x,y,z}. That, or a QgsPointV2::createWithZ(real, real, real) static.